src/main/java/org/apache/sling/commons/metrics/internal/JmxExporterFactory.java

```diff
@@ -78,14 +85,46 @@ public class JmxExporterFactory {
     MetricsService metrics;
     
     MBeanServer server;
-    
+
+    NotificationListener listener = new NotificationListener() {
+        public void handleNotification(Notification notification, Object handback) {
+            if (MBeanServerNotification.REGISTRATION_NOTIFICATION.equals(notification.getType())) {
+                ObjectName objectname = null;
+                try {
+                    if(notification instanceof MBeanServerNotification) {
+                        MBeanServerNotification mbeanNotification = (MBeanServerNotification) notification;
+                        objectname = mbeanNotification.getMBeanName();
+                        LOG.debug("JMX Notification : match {} with pattern = {}", objectname, Arrays.asList(patterns));
+                        for (String pattern : patterns) {
+                            if (objectname.toString().matches(pattern)) {
+                                LOG.debug("JMX Notification : register metrics for MBean: {}", objectname);
+                                registerMBeanProperties(objectname);
+                                break;
+                            }
+                        }
+                    } else {
+                        LOG.debug("JMX Notification : Cannot register metrics for objectname = {}", objectname);
```

Comment:
```objectname```  is still ```null```  here. You rather need 

```suggestion
                        LOG.debug("JMX Notification : Cannot handle notification, because it's not a MBeanServerNotification ({})", notification);
```

Reply:
Changed